pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -1442,18 +1462,7 @@ func (gce *GCECloud) AttachDisk(diskName string, readOnly bool) error {
 
 	attachOp, err := gce.service.Instances.AttachDisk(gce.projectID, gce.zone, gce.instanceID, attachedDisk).Do()
 	if err != nil {
-		// Check if the disk is already attached to this instance.  We do this only
-		// in the error case, since it is expected to be exceptional.
-		instance, err := gce.service.Instances.Get(gce.projectID, gce.zone, gce.instanceID).Do()
-		if err != nil {
-			return err
-		}
-		for _, disk := range instance.Disks {
-			if disk.Source == attachedDisk.Source {
-				// Disk is already attached, we're good to go.
-				return nil
-			}
-		}
+		return err
```

Comment:
Yep, intentional. If the disk is already attached, I want this to error out, and let the PD attach/detach code resolve it.